### DISCUSSION TITLE
How to fix the colorbar range?

### DISCUSSION THREAD

**Discussion by e-aranda** [GLVis/glvis#78] [Q&A] [answered]
I would like to represent different results with the same colorbar range. I mean, 
I have two sets of data, the first one with values between 0 and 1, the second one between 0 and 0.7. Using the same palette, the color of a 0.5 value in the first one data set is different from the color of the second one for the same value because colorbar is adapted to the range values. Is it possible to represent the second set of data with a colorbar going from 0 to 1?

**Comment by jakubcerveny** [comment, marked as answer]
I think it's the `F7` key that you can set the min/max of the colorbar with. If you press `H`, a help on most of the keys is displayed in the terminal.

**Comment by e-aranda** [comment]
Ok, thank you. I did not know it was possible to do it. I also have found how to set using key parameters. In case anyone is interested:

    sout << "valuerange min max\n";

By the way, is there a tutorial where all this option are explained?

**Comment by tzanio** [comment]
The `README` file is probably the best "tutorial" for the available keystrokes. 

All GLVis "commands" like `valuerange` above a listed in this portion of the code: https://github.com/GLVis/glvis/blob/e46168bbcc081277ddff17d6d378e5c561b0c1db/glvis.cpp#L763-L1020

We plan to work on better documentation in the future.

**Comment by v-dobrev** [comment]
Just to clarify: there are two different (though intentionally very similar) sets of GLVis commands: one for commands in GLVis scripts (executed from the command line with the `-run`/`--run-script` option) and commands sent to GLVis through socket connections. What @tzanio linked above are the script commands. The socket commands are in the following big `if` statement: https://github.com/GLVis/glvis/blob/f85cf7c75b19ea59a9cbeb8dc33d5301676d9b01/lib/threads.cpp#L810-L1245

**Comment by v-dobrev** [comment]
By the way, regarding GLVis scripts, there is a short description of this capability here: https://glvis.org/options-and-use.

**Comment by HennesHajduk** [comment]
Using these hints I am able to do one of the following things: 
1) Rescale the socketstream with F7 key for each image. Downside: works only manually.
2) Run a glvs-script to adjust the valuerange before taking a snapshot (or doing something similar). Downside: Output to files is required first.

I wonder if there is an easy way to have the valuerange "pre-defined" when an inputstream is sent live to glvis (when a non-steady problem is solved). I tried

`sock << "solution\n";`
`mesh.Print(sock);`
`gf.Save(sock);`
`sock << "valuerange 0 1\n";`

Apparently, `valuerange` has to be specified after the `GridFunction` is saved. So this approach first plots in the custom range and then re-scales it just afterwards, which makes it hard to work with for me.

Any ideas?

**Comment by v-dobrev** [comment]
@HennesHajduk, there is a command `autoscale <mode>`, where `<mode>` can be:
* `off` - turn off autoscaling for both mesh box and value range
* `on` (default) - turn on autoscaling for both mesh box and value range
* `mesh` - turn on autoscaling just for the mesh box
* `value` - turn on autoscaling just for the value range.

It is available as both: a stream and a script command.
